### PR TITLE
ICU4C lang names working now...

### PR DESCRIPTION
**Pull request by sven-oly**

### CODE REVIEW

File: executors/cpp/main.cpp [hunk 1]
```diff
@@ -109,11 +111,19 @@ int main(int argc, const char** argv)
       // else if (test_type == "number_fmt") {
       //   outputLine = test_numfmt(json_input);
       //   // }
-      // else if (test_type == "language_display_name") {
-      //   outputLine = test_langnames(json_input);
-      // }
-else {
-        std::string outputLine = "# BAD TEST " + test_type;
+      else if (test_type == "lang_names" ||
```
Comment (echeran) [inline executors/cpp/main.cpp:114]: We should really have only one value in this key (b/c it behaves like an enum) to represent this one component type.

File: testdriver/testplan.py [hunk 2]
```diff
@@ -523,15 +523,17 @@ def send_one_line(self, input_line):
                 # TODO!!!! Return an error for the offending line instead of failing for the whole batch
 
                 return None
-                input = json.loads(input_line.replace('#EXIT', ''))
-                error_result = {'label': input['label'],
-                                'input_data': input,
-                                'error': self.run_error_message
-                }
+                try:
```
Comment (echeran) [inline testdriver/testplan.py:526]: insert an empty to visually separate the try/except block

Comment (echeran) [reply testdriver/testplan.py:526]: also: do you even need the try/except block here? see if you can remove it

**Review by echeran — CHANGES_REQUESTED** [review]

File: executors/cpp/main.cpp (hunk 1 above)
Comment (sven-oly) [reply executors/cpp/main.cpp]: Agreed.

File: executors/cpp/main.cpp [hunk 3]
```diff
@@ -41,15 +41,18 @@ using std::string;
 // Test functions
 extern const string test_collator(json_object *json_in);
 // extern const string test_numfmt(json_object *json_in);
-// extern const string test_langnames(json_object *json_in);
+extern const string test_langnames(json_object *json_in);
+
+// !!! extern const string test_likely_subtags(json_object *json_in);
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/main.cpp]: Done

File: executors/cpp/langnames.cpp [hunk 4]
```diff
@@ -29,38 +29,47 @@ using std::string;
...
-  json_object_object_add(return_json,
-                         "result",
-                         json_object_new_string(name.c_str()));
- string return_str = json_object_to_json_string(return_json);
 
+  char test_result_string[1000] = "";
+
+  int32_t chars_out = testLang.extract(test_result_string, 1000, nullptr, status);
+  if (U_FAILURE(status)) {
+    json_object *error_msg = json_object_new_object();
```
[the remark this answers is not in the document]
Comment (sven-oly) [reply executors/cpp/langnames.cpp]: Done

File: testdriver/testplan.py (hunk 2 above)
Comment (sven-oly) [reply testdriver/testplan.py]: Changed this. PTAL.

File: testdriver/testplan.py [hunk 5]
```diff
@@ -520,18 +520,15 @@ def send_one_line(self, input_line):
...
+                error_result = {'label': 'UNKNOWN',
+                                'input_data': input_line,
                                 'error': self.run_error_message
-                }
+                                }
                 return json.dumps(error_result)
         except BaseException as err:
             logging.error('!!! send_one_line fails: input => %s<. Err = %s', input_line, err)
-            input = json.loads(input_line.replace('#EXIT', ''))
+            input = json.loads(input_line.replace('#EXIT', '').strip())
```
Comment (echeran) [inline testdriver/testplan.py:531]: This seems harmless, but why do we need to replace `#EXIT` from the input string when recording the error? In other words, why aren't we faithfully documenting the input line as it is? I wouldn't expect an `#EXIT` line to be the cause of errors, but even if it is, then our logs should be a faithful source of the evidence.

**Review by echeran — APPROVED** [review]
Looks fine, LGTM. But take a look at the question about why we're logging a transformed version of the input during an error scenario (and not the original input itself).

File: testdriver/testplan.py (hunk 5 above)
Comment (sven-oly) [reply testdriver/testplan.py:531]: Let's review this later. Tnx for for the LGTM.

**Pull request merged**